metadata-ingestion/src/datahub/ingestion/source/looker/view_upstream.py

```diff
@@ -236,6 +247,52 @@ def get_upstream_dataset_urn(self) -> List[Urn]:
     def create_fields(self) -> List[ViewField]:
         return []  # it is for the special case
 
+    def create_upstream_column_refs(
+        self, upstream_urn: str, expected_columns: List[str]
+    ) -> List[ColumnRef]:
+        """
+        - **`upstream_urn`**: The URN of the upstream dataset.
+
+        - **`expected_columns`**: These are the columns identified by the Looker connector as belonging to the `upstream_urn` dataset. However, there is potential for human error in specifying the columns of the upstream dataset. For example, a user might declare a column in lowercase, while on the actual platform, it may exist in uppercase, or vice versa.
+
+        - This function ensures consistency in column-level lineage by consulting GMS before creating the final `ColumnRef` instance, avoiding discrepancies.
+        """
+        actual_columns: List[str] = []
+        with SchemaResolver(
+            platform=self.view_context.view_connection.platform,
+            platform_instance=self.view_context.view_connection.platform_instance,
+            env=self.view_context.view_connection.platform_env or self.config.env,
+            graph=self.ctx.graph,
+        ) as schema_resolver:
+            urn, schema_info = _get_schema_info(schema_resolver, upstream_urn)
+
+            if schema_info:
+                column_from_gms: List[str] = list(
+                    schema_info.keys()
+                )  # list() to silent lint
+                actual_columns = [
+                    column
+                    for column in column_from_gms
+                    if column.lower() in map(str.lower, expected_columns)
```

Comment:
would be good to extract this logic out into a helper method

Reply:
done